lib/cancan/ability.rb

```diff
@@ -369,9 +366,30 @@ def relevant_rules(action, subject)
         rule.relevant? action, subject
       end
       relevant.reverse!.uniq!
+      optimize_order! relevant
       relevant
     end
 
+    # Optimizes the order of the rules, so that rules with the :all subject are evaluated first.
+    def optimize_order!(rules)
+      fst_pos_i = -1
+      rules.each_with_index do |rule, i|
+        unless rule.base_behavior
```

Comment:
can we inline this doing something like `first_position_index = -1 and next unless rule.base_behaviour`?